include/lsst/meas/extensions/shapeHSM/HsmMomentsControl.h

```diff
@@ -158,25 +196,81 @@ class HsmSourceMomentsAlgorithm : public HsmMomentsAlgorithm {
     Control _ctrl;
 };
 
+
 /// Class to measure HSM adaptive moments of PSF
 class HsmPsfMomentsAlgorithm : public HsmMomentsAlgorithm {
 public:
     /// A typedef to the Control object for this algorithm, defined above.
     ///  The control object contains the configuration parameters for this algorithm.
-    typedef HsmPsfMomentsControl Control;
+    typedef std::shared_ptr<const HsmPsfMomentsControl> Control;
 
     /// @brief Initialize with standard field names and customized documentation.
     HsmPsfMomentsAlgorithm(Control const & ctrl, std::string const & name, afw::table::Schema & schema) :
         HsmMomentsAlgorithm(name, schema, "Psf adaptive moments algorithm from HSM"), _ctrl(ctrl) {}
 
+    virtual ~HsmPsfMomentsAlgorithm() {}
+
     void measure(
         afw::table::SourceRecord & measRecord,
         afw::image::Exposure<float> const & exposure
     ) const;
 
-private:
+protected:
+    virtual std::shared_ptr<afw::detection::Psf::Image> getPsfImage(
+        geom::Point2D center,
+        afw::table::SourceRecord & source,
+        afw::image::Exposure<float> const & exposure
+    ) const;
+
+    virtual std::shared_ptr<afw::image::Mask<afw::image::MaskPixel>> getPsfMask(
+        geom::Point2D center,
+        afw::table::SourceRecord & source,
+        afw::image::Exposure<float> const & exposure
+    ) const;
+
+    virtual afw::image::MaskPixel const getBadPixelMask(
+        afw::image::Exposure<float> const & exposure
+    ) const { return 0; }
+
     Control _ctrl;
 };
+
+
+/// Measure PSF moments while making image signal-to-noise look similar to underlying star,
+/// thereby removing any signal-to-noise related biases.
+class HsmPsfMomentsDebiasedAlgorithm : public HsmPsfMomentsAlgorithm {
+public:
+    static base::FlagDefinition const EDGE;
+
+    typedef std::shared_ptr<const HsmPsfMomentsDebiasedControl> Control;
+
+    HsmPsfMomentsDebiasedAlgorithm(Control const & ctrl, std::string const & name, afw::table::Schema & schema) :
+        HsmPsfMomentsAlgorithm(ctrl, name, schema) {
+            auto thisCtrl = std::static_pointer_cast<Control::element_type>(_ctrl);
+            if ((thisCtrl->noiseSource != "meta") && (thisCtrl->noiseSource != "variance")) {
+                throw LSST_EXCEPT(base::MeasurementError, "invalid noiseSorce", FAILURE.number);
+            }
+        }
+
+    virtual ~HsmPsfMomentsDebiasedAlgorithm() {}
+
+private:
+    virtual std::shared_ptr<afw::detection::Psf::Image> getPsfImage(
+        geom::Point2D center,
+        afw::table::SourceRecord & source,
+        afw::image::Exposure<float> const & exposure
+    ) const override;
```

Comment:
Please drop the `virtual` from this and the methods below it: https://developer.lsst.io/cpp/style.html#the-override-specifier-should-be-used-whenever-the-intention-is-to-override-a-function-from-a-base-class